### PR TITLE
Host refactor 

### PR DESCRIPTION
**Pull request by jajanusz**
For #1229 

Please consider it a draft - feel free to write what you'd do differently.

Details regarding some bigger steps are in commit messages.
The aim of this PR is to simplify FW tree structure and make code less coupled.
Testbench is going to be a part of tools and just use library produced by the FW buildsystem.

`library` folder with platform headers was moved to `platform` so library is placed alongside other platforms, it's like any other platform (even has kconfig for it now), just abstract, but IMHO it's better to have it here instead of having it in src root.

I made FW build simpler, but testbench build may seem a bit harder because you have to export sof library project to use it, what may look scary for someone that doesn't know CMake - we have only 1 app that uses sof lib ATM, I may add some utility function to make it simpler when we will have more, or just create some kind of framework for these apps.
Anyway I think it's good change because we should care first about what's our main focus - Firmware, testbench is still easy to build, just CMake file related to it got more complicated, but they probably will not be touched often.

Commits are arranged in the order that doesn't break `git bisect`.

### CODE REVIEW

File: src/arch/host/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ set(GENERATED_DIRECTORY ${PROJECT_BINARY_DIR}/generated)
 file(MAKE_DIRECTORY ${GENERATED_DIRECTORY}/include)
 
 target_include_directories(sof_options INTERFACE ${PROJECT_SOURCE_DIR}/src/arch/host/include)
-target_include_directories(sof_options INTERFACE ${PROJECT_SOURCE_DIR}/src/library/include)
+target_include_directories(sof_options INTERFACE ${PROJECT_SOURCE_DIR}/src/platform/library/include)
```
Comment (lgirdwood) [inline src/arch/host/CMakeLists.txt]: Please say why this is better in commit message.

Comment (jajanusz) [reply src/arch/host/CMakeLists.txt]: Done

**Comment by jajanusz** [comment]
@singalsu @mmaka1 @tlauda ping

**Comment by lgirdwood** [comment]
@jajanusz CI UT's failing btw.

**Comment by jajanusz** [comment]
> @jajanusz CI UT's failing btw.

It's about python tests and it's false negative, cos one of platforms in CI is down... 
Idk if they gonna bring it up before monday

**Comment by lgirdwood** [comment]
@jajanusz and CI build failing too.

**Comment by jajanusz** [comment]
> Looks great, just small typo in third commit message: "int" -> "in".

Fixed. Thanks

**Comment by jajanusz** [comment]
@xiulipan Why is `sof-ci/jenkins/pr-fw-build` pending for like 5 hours? I see there are logs etc, but has some problems with publishing results.

**Comment by xiulipan** [comment]
@jajanusz 
Must be some network error with GitHub REST API. It happens.
I will re-trigger the test.

**Pull request merged**